spec/helper/get_docker_params_changed.rb

```diff
@@ -0,0 +1,177 @@
+      pp_mounts = opts['volumes'].select { |item| item.include?(':') } if opts['volumes'].is_a?(Array) && opts['osfamily'] != 'windows'
+      pp_paths = opts['volumes'].select { |item| item.scan(%r{(?=:)}).count == 1 } if opts['volumes'].is_a?(Array) && opts['osfamily'] == 'windows'
+      pp_mounts = opts['volumes'].select { |item| item.scan(%r{(?=:)}).count == 2 } if opts['volumes'].is_a?(Array) && opts['osfamily'] == 'windows'
+
+      inspect_paths = if inspect_hash['Config']['Volumes']
+                        inspect_hash['Config']['Volumes'].keys
+                      else
+                        []
+                      end
+      param_changed = true if pp_paths != inspect_paths
+
+      names = inspect_hash['Mounts'].map { |item| item.values[1] } if inspect_hash['Mounts']
+      pp_names = pp_mounts.map { |item| item.split(':')[0] } if pp_mounts
+      names = names.select { |item| pp_names.include?(item) } if names && pp_names
+      destinations = inspect_hash['Mounts'].map { |item| item.values[3] } if inspect_hash['Mounts']
+      pp_destinations = pp_mounts.map { |item| item.split(':')[1] } if pp_mounts && opts['osfamily'] != 'windows'
+      pp_destinations = pp_mounts.map { |item| "#{item.split(':')[1].downcase}:#{item.split(':')[2]}" } if pp_mounts && opts['osfamily'] == 'windows'
+      destinations = destinations.select { |item| pp_destinations.include?(item) } if destinations && pp_destinations
+
+      param_changed = true if pp_names != names
+      param_changed = true if pp_destinations != destinations
+      param_changed = true if pp_mounts != [] && inspect_hash['Mounts'].nil?
+
+      # check if something on ports was changed(some ports were added or removed)
+
+      ports = inspect_hash['HostConfig']['PortBindings'].keys
+      ports = ports.map { |item| item.split('/')[0] }
+      pp_ports = opts['ports'].sort if opts['ports'].is_a?(Array)
+      pp_ports = [opts['ports']] if opts['ports'].is_a?(String)
+
+      param_changed = true if pp_ports && pp_ports != ports
+
+      if param_changed
+        remove_container(opts['sanitised_title'], opts['osfamily'], opts['stop_wait_time'], opts['cidfile'], log)
+        create_container(opts['command'], opts['osfamily'], opts['image'], log)
+        return_value = 'PARAM_CHANGED'
+        # log.puts(return_value)
+      end
+    else
+      create_container(opts['command'], opts['osfamily'], opts['image']) unless File.exist?(opts['cidfile'])
+      _stdout, _stderr, status = Open3.capture3("docker inspect #{opts['sanitised_title']}")
+      # log.puts("command: #{"docker inspect #{opts['sanitised_title']}"}\nstdout: #{stdout}\nstderr: #{stderr}\nstatus: #{status}\n\n")
+      unless status.to_s.include?('exit 0')
+        remove_cidfile(opts['cidfile'], opts['osfamily'])
+        create_container(opts['command'], opts['osfamily'], opts['image'])
+      end
+      return_value = 'NO_CHANGE_DETECTED'
```

Comment:
Would perhaps suggest:
```suggestion
      return_value = 'No changes detected'
```
...to keep it consistent with the rest of the output